shared/components/detailView.js

```diff
@@ -22,30 +23,43 @@ const styles = {
   }
 };
 
-export default class detailView extends Component {
+export default class DetailView extends Component {
 
   constructor(props) {
     super(props);
   }
 
+  componentDidMount() {
+    this._swiper.scrollBy(this.props.rowID * 1);
+  }
+
+
   render() {
     return (
       <View>
-        <Swiper style={styles.wrapper} showsButtons>
-          {this.props.detailSource.map((value)=>{
-            return (
-              <View style={styles.slide}>
-                <Image source = {{uri: value.url}}
-                       style = {styles.image}/>
-              </View>
-            );
-          })}
-        </Swiper>
+        <View>
+          <DetailHeader title = {this.props.title}/>
+        </View>
+        <View>
+          <Swiper style={styles.wrapper} showsButtons ref={(swiper) => {this._swiper = swiper;}} loop = {false}>
+            {this.props.detailSource.map((value)=>{
+              return (
+                <View style={styles.slide}>
+                  <Image source = {{uri: value.url}}
+                         style = {styles.image}/>
+                </View>
+              );
+            })}
+          </Swiper>
+        </View>
       </View>
     );
   }
 }
 
-detailView.propTypes = {
-  detailSource: PropTypes.any
+DetailView.propTypes = {
+  detailSource: PropTypes.any,
+  detailIndex: PropTypes.any,
```

Comment:
I don't see detailIndex prop being used in this component anywhere. please delete un-used props.